Disable CSV parsing by default and update tests to better show what is left (#2072)
NVIDIA/spark-rapids: docs/compatibility.md

## CSV Reading

Due to inconsistencies between how CSV data is parsed CSV parsing is off by default.
Each data type can be enabled or disabled independently using the following configs.

 * [spark.rapids.sql.csv.read.bool.enabled](configs.md#sql.csv.read.bool.enabled)
 * [spark.rapids.sql.csv.read.byte.enabled](configs.md#sql.csv.read.byte.enabled)
 * [spark.rapids.sql.csv.read.date.enabled](configs.md#sql.csv.read.date.enabled)
 * [spark.rapids.sql.csv.read.double.enabled](configs.md#sql.csv.read.double.enabled)
 * [spark.rapids.sql.csv.read.float.enabled](configs.md#sql.csv.read.float.enabled)
 * [spark.rapids.sql.csv.read.integer.enabled](configs.md#sql.csv.read.integer.enabled)
 * [spark.rapids.sql.csv.read.long.enabled](configs.md#sql.csv.read.long.enabled)
 * [spark.rapids.sql.csv.read.short.enabled](configs.md#sql.csv.read.short.enabled)
 * [spark.rapids.sql.csvTimestamps.enabled](configs.md#sql.csvTimestamps.enabled)

If you know that your particular data type will be parsed correctly enough, you may enable each

> type you expect to use. Often the performance improvement is so good that it is worth dealing

Reviewer: ```suggestion
type you expect to use. Often the performance improvement is so good that it is worth
```

checking if it is parsed correctly.

Spark is generally very strict when reading CSV and if the data does not conform with the 

> expected format exactly it will result in a `null` value. The underlying parser that the SQL plugin

Reviewer: ```suggestion
expected format exactly it will result in a `null` value. The underlying parser that the RAPIDS Accelerator
```

As a work around you can parse the column as a timestamp and then cast it to a date.

> Invalid dates, ones that have the correct format, but the numbers produce invalid dates, in Spark
> can result in an exception by default and how they are parsed can be controlled through a config.

Reviewer: ```suggestion
Invalid dates in Spark, values that have the correct format but the numbers produce invalid dates,
can result in an exception by default, and how they are parsed can be controlled through a config.
```

> The plugin does not support any of this and will produce an incorrect date.  Typically, one that

Reviewer: ```suggestion
The RAPIDS Accelerator does not support any of this and will produce an incorrect date.  Typically, one that
```

Just like with dates all timestamp formats are actually supported at the same time.  The plugin will
disable itself if it sees a format it cannot support.

> Invalid timestamps, ones that have the correct format, but the numbers produce invalid dates or

Reviewer: Same rewording suggestions as above

Any number that overflows will not be turned into a null value.

Also parsing of some values will not produce bit for bit identical results to what the CPU does.

> they are within round-off errors except when they are close enough to overflow to Inf or -Inf which

Reviewer: ```suggestion
They are within round-off errors except when they are close enough to overflow to Inf or -Inf which
```

> then results in a number being returned when the CPU would have returns null.

Reviewer: ```suggestion
then results in a number being returned when the CPU would have returned null.
```
